source/Government.cpp

```diff
@@ -32,31 +32,57 @@ using namespace std;
 
 namespace {
 	// Load ShipEvent strings and corresponding numerical values into a map.
-	void PenaltyHelper(const DataNode &node, map<int, double> &penalties)
+	void PenaltyHelper(const DataNode &node, map<int, Government::PenaltyEffect> &penalties)
 	{
+		auto loadPenalty = [&penalties](const DataNode &child, int eventType) -> void
+		{
+			double amount = child.Value(1);
+			Government::SpecialPenalty specialPenalty = Government::SpecialPenalty::NONE;
+			if(child.Size() >= 3)
+			{
+				const string &effect = child.Token(2);
+				if(effect == "none")
+					specialPenalty = Government::SpecialPenalty::NONE;
+				else if(effect == "provoke")
+					specialPenalty = Government::SpecialPenalty::PROVOKE;
+				else if(effect == "atrocity")
+				{
+					specialPenalty = Government::SpecialPenalty::ATROCITY;
+					if(amount <= 0.)
+					{
+						child.PrintTrace("Warning: the \"atrocity\" effect will not work"
+							" without a positive, non-zero penalty to reputation, defaulting to 0.1:");
+						amount = .1;
+					}
+				}
+				else
+					child.PrintTrace("Skipping unrecognized reputation effect:");
+			}
+			penalties[eventType] = Government::PenaltyEffect(amount, specialPenalty);
+		};
```

Comment:
If someone sets a custom penalty for the `provoke` or `atrocity` events with just a number, it'll discard the `PROVOKE` or `ATROCITY` `SpecialPenalty` you've included, which is problematic because now provoking someone isn't actually going to provoke them.